lib/payrix/client.rb

```diff
@@ -4,7 +4,7 @@ module Payrix
   # The Payrix::Client is used both internally and externally to initiate API requests to the Payrix API.
   class Client
     def request(method:, resource:, data: {}, filters: {}, options: {})
-      url = Payrix.configuration.url
+      url = Payrix.configuration.url(options[:test_mode])
```

Comment:
This is the developer-facing change. Here we accept the new `:test_mode` parameter, and pass it into `Payrix::Configuration#url` to set the environment used per-request.